File: Homework/HW5/led.py
#!/usr/bin/python3
import sys
from time import sleep
LED_Path = "/sys/class/gpio/gpio17/"
sysfs_dir = "/sys/class/gpio/"
LED_number = "17"

def writeLED( filename, value, path = LED_Path ):
	fo = open( path + filename, "w")
	fo.write(value)
	fo.close()
	return
	
#This section sets up the GPIO Pin for the program
writeLED(filename = "export", value = LED_number, path = sysfs_dir)
sleep(0.1)
writeLED(filename = "direction", value = "out")

#This section blinks the LED at 0.5s intervals
for x in range(0,20):
	writeLED(filename = "value", value = "1")
	sleep(0.5)
	writeLED(filename = "value", value = "0")
	sleep(0.5)

#This section closes down the operation
writeLED(filename = "unexport", value = LED_number, path = sysfs_dir)

File: Homework/HW5/led.c
#include<stdio.h>
#include<unistd.h>
#include<stdlib.h>
#include<string.h>

#define GPIO_NUMBER "17"
#define GPIO_PATH "/sys/class/gpio/gpio17/"
#define GPIO_SYSFS "/sys/class/gpio/"

void writeGPIO(char filename[], char value[]) {
	FILE* fp;
	fp = fopen(filename, "w+");
	fprintf(fp, "%s", value);
	fclose(fp);
}

int main(){
	/*Setting up the LED pin for use*/
	writeGPIO(GPIO_SYSFS "export", GPIO_NUMBER);
	sleep(1);
	writeGPIO(GPIO_PATH "direction", "out");
	/*Blinking the LED at 0.5s intervals*/
	for (int a = 0; a < 20; a = a +1){
		writeGPIO(GPIO_PATH "value", "1");
		sleep(.5);
		writeGPIO(GPIO_PATH "value", "0");	
		sleep(.5);
	}
	/*Shutting down the LED pin*/
	writeGPIO(GPIO_SYSFS "unexport", GPIO_NUMBER);
	return 0;
}
